src/FileFormat/PdfStatistics.php

```diff
@@ -0,0 +1,217 @@
+        self::PAGE_FORMAT_4A0_SIZE => '4A0',
+        self::PAGE_FORMAT_2A0_SIZE => '2A0',
+        self::PAGE_FORMAT_A0_SIZE => 'A0',
+        self::PAGE_FORMAT_A1_SIZE => 'A1',
+        self::PAGE_FORMAT_A2_SIZE => 'A2',
+        self::PAGE_FORMAT_A3_SIZE => 'A3',
+        self::PAGE_FORMAT_A4_SIZE => 'A4',
+        self::PAGE_FORMAT_A5_SIZE => 'A5',
+        self::PAGE_FORMAT_A6_SIZE => 'A6',
+        self::PAGE_FORMAT_A7_SIZE => 'A7',
+        self::PAGE_FORMAT_A8_SIZE => 'A8',
+        self::PAGE_FORMAT_A9_SIZE => 'A9',
+        self::PAGE_FORMAT_A10_SIZE => 'A10',
+    );
+
+    /**
+     * @var \Imagick Imagick object
+     */
+    protected $imagickDocument;
+
+    /**
+     * Constructor checks, if Imagick extension is available. If yes, instance is created.
+     * @throws \RuntimeException if Imagick extension is not available
+     */
+    public function __construct()
+    {
+        if (!extension_loaded('imagick')) {
+            new \RuntimeException('Imagick php extension is needed to perform PDF statistics.');
+        }
+
+        $this->imagickDocument = new \Imagick();
+    }
+
+    /**
+     * Set filename.
+     * @param string $filename path of pdf file to be analysed
+     */
+    public function setFilename($filename)
+    {
+        $this->imagickDocument->clear();
+        $this->imagickDocument->setResolution(72, 72);
+        $this->imagickDocument->readImage($filename);
+        $this->imagickDocument->setImageUnits(\Imagick::RESOLUTION_PIXELSPERINCH);
+    }
+
+    /**
+     * Method checks, if pdf document pages are in color.
+     * @return array each element is index of page, which is in color (starts from 0)
+     */
+    public function getPagesInColor()
+    {
+        $pagesInColor = array();
+        $pageIterator = 0;
+
+        foreach ($this->imagickDocument as $singlePage) {
+            $histogram = $singlePage->getImageHistogram();
+
+            foreach ($histogram as $pixel) {
+                if (!(
+                    ($pixel->getColorValue(\Imagick::COLOR_RED)==$pixel->getColorValue(\Imagick::COLOR_GREEN)) && 
+                    ($pixel->getColorValue(\Imagick::COLOR_GREEN)==$pixel->getColorValue(\Imagick::COLOR_BLUE)) 
+                )) {    // grayscale is only, when rgb values are the same
+                    $pagesInColor[] = $pageIterator;
+                    break;
+                }
+            }
+
+            $pageIterator++;
+        }
+
+        return $pagesInColor;
+    }
+
+    /**
+     * Function return number of pages in pdf document.
+     * @return integer number of pages
+     */
+    public function getNumberOfPages()
+    {
+        return $this->imagickDocument->getNumberImages();
+    }
+
+    /**
+     * Method identifying page format.
+     * @param string $pageSize page size in 1x1 format
+     * @param boolean $returnPageSizeInMmIfNotMatched if set to false and page size does not fit then function return false (default true)
```

Comment:
Hint: according to our style guide (based on [symfony style](https://symfony.com/doc/current/contributing/code/standards.html)), we write `bool` (and `int`). And put an empty line (exactly with * only) before and after @param group. (As said, hint only).

generally: run composer install to install the development dependencies. This will install a pre-commit hook. (And then run this one manually before or after committing with netbeans.)